arrow-udf-flight/src/lib.rs

```diff
@@ -45,9 +45,10 @@ impl Client {
 
     /// Create a new client.
     pub async fn new(channel: Channel) -> Result<Self> {
-        let mut client = FlightServiceClient::new(channel);
+        Self::with_flight_client(FlightServiceClient::new(channel)).await
+    }
 
-        // get protocol version in server
+    pub async fn with_flight_client(mut client: FlightServiceClient<Channel>) -> Result<Self> {
```

Comment:
Do you think it's a good idea to just change `fn new(channel: Channel)` into `fb new(FlightServiceClient<Channel>)`?

Reply:
But that'll break the API. I'm personally OK with changing signature of `new`.

Reply:
> But that'll break the API. I'm personally OK with changing signature of `new`.

Our next release will update arrow, so introducing breaking changes is acceptable. I prefer using `new(FlightServiceClient<Channel>)` instead.